packages/sdk/src/services/holograph-protocol.service.ts

```diff
@@ -92,4 +103,98 @@ export class HolographProtocol {
   get holographDropERC721(): HolographDropERC721 {
     return this.holographDropERC721Contract
   }
+
+  async hydrateContractFromAddress(hydrateContractInput: {
+    chainId: number
+    address: Address
+    type: CollectionType
+  }): Promise<HolographLegacyCollection | HolographMoeERC721DropV1 | HolographMoeERC721DropV2> {
+    const {chainId, address, type} = hydrateContractInput
+
+    const abi = parseAbi([
+      'function name() external view returns (string memory)',
+      'function symbol() external view returns (string memory)',
+      'function contractURI() external view returns (string memory)',
+      'function config() external view returns (Configuration)',
+      'struct Configuration { uint160 metadataRenderer;  uint64 editionSize; uint16 royaltyBPS; address fundsRecipient;}',
+      'function salesConfig() external view returns (SalesConfiguration)',
+      'struct SalesConfiguration { uint104 publicSalePrice; uint32 maxSalePurchasePerAddress; uint64 publicSaleStart; uint64 publicSaleEnd; uint64 presaleStart; uint64 presaleEnd; bytes32 presaleMerkleRoot; }',
+    ])
+
+    const client = {public: this._providers.byChainId(chainId)}
```

Comment:
@BravoNatalie This may have to be another ticket, but we have to handle the case when someone enters an invalid chainId, or contract type.